web/src/components/projects/CreateProjectForm.tsx

```diff
@@ -436,81 +474,144 @@ export default function CreateProjectForm({ approvedStudies, handleProjectCreate
+                                      getValues(`additionalApprovedResearchers.${index}.roles`) || [];
+                                    setValue(`additionalApprovedResearchers.${index}.roles`, [
+                                      ...currentRoles,
+                                      e.target.value as AnyProjectRoleName,
+                                    ]);
+                                  }
+                                }}
                                 disabled={isSubmitting}
-                              />
-                              <span className={styles["domain-suffix"]}>{domainName}</span>
-                            </div>
-                            {fieldState.error && (
-                              <Alert type="error">
-                                <AlertMessage>{fieldState.error.message}</AlertMessage>
-                              </Alert>
+                              >
+                                <option value="">+ Add role...</option>
+                                {availableRolesToAdd.map((role) => (
+                                  <option key={role} value={role}>
+                                    {ROLE_LABELS[role]}
+                                  </option>
+                                ))}
+                              </select>
                             )}
                           </div>
-                        )}
-                      />
-
-                      <button
-                        type="button"
-                        onClick={() => removeResearcher(index)}
-                        className={styles["remove-button"]}
-                        aria-label={`Remove researcher ${index + 1}`}
-                      >
-                        ×
-                      </button>
-                    </div>
-                  ))}
+                        </div>
+
+                        <button
+                          type="button"
+                          onClick={() => removeResearcher(index)}
+                          className={styles["remove-button"]}
+                          aria-label={`Remove researcher ${index + 1}`}
+                        >
+                          ×
+                        </button>
+                      </div>
+                    );
+                  })}
 
                   <Button
                     className={styles["add-button"]}
                     type="button"
                     variant="secondary"
                     size="small"
-                    onClick={() => appendResearcher({ value: "" })}
+                    onClick={() => appendResearcher({ username: "", roles: [] })}
                   >
                     Add Researcher
                   </Button>
                 </fieldset>
-                <HelperText>Optionally add approved researchers to this project</HelperText>
+                <HelperText>Optionally add approved researchers with their roles to this project</HelperText>
```

Comment:
to be an uploader you don't need to be an approved researcher

Reply:
is there any requirement to be staff for any of the roles?